storage/series/series.go

```diff
@@ -155,8 +155,16 @@ func (s *dbSeries) needsBlockUpdateWithRLock() bool {
 }
 
 func (s *dbSeries) shouldExpire(now, blockStart time.Time) bool {
+	var cutoff time.Time
 	rops := s.opts.RetentionOptions()
-	cutoff := now.Add(-rops.RetentionPeriod()).Truncate(rops.BlockSize())
+	if rops.DiskMode() {
+		// Some blocks of series is stored in disk only
+		cutoff = now.Add(-rops.DiskModeInMemPeriod()).Truncate(rops.BlockSize())
+	} else {
+		// Everything for the retention period is kept in mem
+		cutoff = now.Add(-rops.RetentionPeriod()).Truncate(rops.BlockSize())
+	}
```

Comment:
This won't be extremely useful - you can achieve the same thing by just changing the retention period shorter on the nodes you are running.  We don't need an extra option to achieve that.

To achieve what you want, which is keeping the files around longer - it would be better to simply add a file cleanup offset so that it expires and writes to disk but then doesn't cleanup until it's older than `- retentionPeriod - cleanupFilesRetentionOffset`.